### PR TITLE
Implement methods in JourneyHandler.

### PR DESCRIPTION
**Pull request by ak47na**

### CODE REVIEW

**Review by anamariamilcu — APPROVED** [review]

File: src/main/java/com/google/sps/data/Journey.java [hunk 1]
```diff
@@ -112,6 +112,11 @@ public boolean findJourneyForTimeslot(DeliverySlot deliverySlot) throws ApiExcep
     ArrayList<ArrayList<Pair>> bestTime = findOptimalTimePaths();
     ArrayList<CourierStop> orderedWaypoints = new ArrayList<>();
     int numberOfWaypoints = getNumberOfWaypoints();
+    
+    if (numberOfWaypoints == 0) {
```
Comment (alexcepoi) [inline src/main/java/com/google/sps/data/Journey.java:116]: This change is fine. But why is this here?

If it was a bug in this code previously, shouldn't this have been caught by a test? If there was no test, shouldn't we add one?

File: src/main/java/com/google/sps/data/JourneyHandler.java [hunk 2]
```diff
@@ -32,17 +32,41 @@
...
+    addDeliveryJourney(journey, deliverySlot);
+    return true;
   }
 
   /**
    * Adds the details of the delivery journey assigned to deliverySlot to datastore.
    */
-  private void addDeliveryJourney(Journey journey, DeliverySlot deliverySlot) throws ApiException, DataNotFoundException, IOException, InterruptedException {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  protected void addDeliveryJourney(Journey journey, DeliverySlot deliverySlot) throws ApiException, BadRequestException, DataNotFoundException, EntityNotFoundException, IOException, InterruptedException {
```
Comment (alexcepoi) [inline src/main/java/com/google/sps/data/JourneyHandler.java]: Why did we change this from protected to private?

File: src/main/java/com/google/sps/servlets/NewRequestServlet.java [hunk 3]
```diff
@@ -160,9 +160,10 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     slotManager.createDeliverySlot(deliverySlot);
     markUserAsCourier(userId);
     JourneyHandler journeyHandler = new JourneyHandler(new GoogleMapsPathFinder());
+    boolean validDeliverySlot;
     try {
       // Create journey for deliverySlot and add it to datastore.
-      journeyHandler.processDeliveryRequest(deliverySlot);
+      validDeliverySlot = journeyHandler.processDeliveryRequest(deliverySlot);
```
Comment (alexcepoi) [inline src/main/java/com/google/sps/servlets/NewRequestServlet.java]: This doesn't seem used anywhere? Why did you bother adding it here?

The delivery slot is super vaild, we it's just that all our existing orders take a much longer time than the user has. This should probably return some sort of error to the user (feel free to TODO it away).

**Review by alexcepoi — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/NewRequestServlet.java (hunk 3 above)
Comment (ak47na) [reply src/main/java/com/google/sps/servlets/NewRequestServlet.java]: It's not used yet, I removed the variable and added a TODO to handle the case.

File: src/main/java/com/google/sps/data/JourneyHandler.java (hunk 2 above)
Comment (ak47na) [reply src/main/java/com/google/sps/data/JourneyHandler.java]: I forgot to update the file. I changed the method to private.

File: src/main/java/com/google/sps/data/Journey.java (hunk 1 above)
Comment (ak47na) [reply src/main/java/com/google/sps/data/Journey.java:116]: It was not a bug, but I added now a test for a journey with no waypoints.
